display_list/geometry/dl_path.h

```diff
@@ -0,0 +1,49 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_DISPLAY_LIST_GEOMETRY_DL_PATH_H_
+#define FLUTTER_DISPLAY_LIST_GEOMETRY_DL_PATH_H_
+
+#include "flutter/display_list/geometry/dl_geometry_types.h"
+#include "flutter/impeller/geometry/path.h"
+#include "flutter/third_party/skia/include/core/SkPath.h"
+
+namespace flutter {
+
+class DlPath {
+ public:
+  DlPath() = default;
+  explicit DlPath(const SkPath& path) : sk_path_(path) {}
+
+  DlPath(const DlPath& path) = default;
+  DlPath(DlPath&& path) = default;
+
+  const SkPath& GetSkPath() const;
+  impeller::Path GetPath() const;
+
+  bool IsInverseFillType() const;
+
+  bool IsRect(DlRect* rect, bool* is_closed = nullptr) const;
+  bool IsOval(DlRect* bounds) const;
+
+  bool IsSkRect(SkRect* rect, bool* is_closed = nullptr) const;
+  bool IsSkOval(SkRect* bounds) const;
+  bool IsSkRRect(SkRRect* rrect) const;
+
+  SkRect GetSkBounds() const;
+  DlRect GetBounds() const;
+
+  bool operator==(const DlPath& other) const;
+
+ private:
+  const SkPath sk_path_;
+  mutable impeller::Path path_;
```

Comment:
does the default initialized impeller::Path end up constructing a shared_ptr even if its not used? Do we need to make this Optional?

Reply:
I'll look into it. I also want to see what happens when you copy construct these objects - does it share the underlying data at all? I believe SkPath will not share the object, but will share the vector data. Not sure about Impeller.

Reply:
IIRC PathRef should behave a bit like a shared_ptr